geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/utils/PathCache.scala

```diff
@@ -98,7 +103,12 @@ object PathCache {
     * @param dir directory path
     * @return
     */
-  def list(fc: FileContext, dir: Path): Iterator[FileStatus] = listCache.get((fc, dir)).iterator
+  def list(fc: FileContext, dir: Path, reload: Boolean = false): Iterator[FileStatus] = {
```

Comment:
Hmm.... neat.  This is a clear example of a function which accepts a boolean that controls whether or not there's a side effect and potentially any threads-safety issues.

Not sure I've see such a clear example of that.  (I'm not necessarily suggesting a change here.)

Reply:
there shouldn't be any thread safety issues b/c everything is going back to guava caches, which are thread safe

Reply:
Yeah, I wasn't thinking that there were any particular thread safety issues; good to know that you've thought through it!